tests/upstream_switch/case.sh

```diff
@@ -217,6 +227,48 @@ function test() {
 	gen_incr_data
 	verify_result
 	clean_task
+	echo "CASE=test_relay success"
+}
+
+function test_binlog_filename_change_when_enable_gtid() {
+	cleanup_process
+	check_master
+	clean_data
+	setup_replica
+	gen_full_data
+	run_dm_components_and_create_sources
+	start_task
+
+	# flush mysql8 master binlog files -> mysql-bin.000002
+	exec_sql $host1 "flush logs;"
+	# flush mysql8 master binlog files -> mysql-bin.000004
+	exec_sql $host2 "flush logs;"
+	exec_sql $host2 "flush logs;"
+	exec_sql $host2 "flush logs;"
+
+	# make mysql8 master slave switch
+	docker-compose -f $CUR/docker-compose.yml pause mysql8_master
+	wait_mysql $host2 2
+
+	# make a flush job to dm
+	exec_sql $host1 "alter table ${db}.${tb} add column c int;"
+	exec_sql $host2 "alter table ${db}.${tb} add column c int;"
+
+	check_sync_diff $WORK_DIR $CUR/conf/diff_config.toml
+	# mysql8 new master's binlog file still is mysql-bin.000001
+	run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"query-status task_pessimistic " \
+		"mysql-bin.000001" 2
```

Comment:
already flushed to 000002?

Reply:
after one switch, source-2 binlog still is 00001

but source-1's binlog is 00002